### PR TITLE
Fix #4871 selfhosted sign in errors

### LINKED ISSUE #4871: FluxC: since develop merge, sign in errors on self hosted site seems broken
Since the latest merge with develop, sign in errors on self hosted sites re-open the .com sign in screen.

### PR DESCRIPTION
**Pull request by maxme**
Fix #4871: user now stays on the correct screen after a sign in error with a self hosted site (+ some cleanup)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 1]
```diff
@@ -501,26 +501,21 @@ protected void finishCurrentActivity(final List<Map<String, Object>> userBlogLis
...
-                        // move on the the main activity
-                        Intent intent = new Intent(getActivity(), WPMainActivity.class);
-                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
-                        intent.putExtra(SignInActivity.MAGIC_LOGIN, true);
-
-                        getActivity().startActivity(intent);
-                    }
 
-                }
+        if (userBlogList != null) {
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: The `finishCurrentActivity(final List<Map<String, Object>> userBlogList)` method this code belongs to seems to be unused - can we drop it completely?

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Hmmm, good point I guess this came from a merge with develop, I'll double check.

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: I think it was introduced when the patch to "clear the url field on sign in" was merged. Fixed in fcbda4e and 61a355d

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**